pkg/reconciler/apiserversource/controller.go

```diff
@@ -59,6 +61,9 @@ func NewController(
 	namespaceInformer := namespace.Get(ctx)
 	serviceaccountInformer := serviceaccountinformer.Get(ctx)
 
+	roleInformer := roleinformer.Get(ctx)
+	rolebindingInformer := rolebindinginformer.Get(ctx)
```

Comment:
Don't we need handlers on those informers when objects we're interested in change ?

Reply:
https://github.com/knative/eventing/pull/7452/files#diff-7d041ee1c1a7f7c9d8479ab20f497e7953367bb9f7666e62dc03f9cbab729f2dR432-R437

https://github.com/knative/eventing/pull/7452/files#diff-7d041ee1c1a7f7c9d8479ab20f497e7953367bb9f7666e62dc03f9cbab729f2dR392-R396

We catch the update here and update it accordingly. Do we need to do that here by adding the handler again? @pierDipi

Reply:
Hey @Leo6Leo,
I think @pierDipi meant the case, when the Role / Rolebinding resource gets changed (e.g. by a user manually). Currently we only "fix" this on the next reconcilation of an ApiServerSource. But it would be good to add an eventhandler for the role & rolebindings to trigger a reconcile of the apiserversource, when a role/rolebinding changes.

E.g. in the ApiServerSource controller we, add an eventhandler for the deployments, which have an owner reference to an ApiServerSource (`FilterFunc: controller.FilterController(&v1.ApiServerSource{})`):
https://github.com/knative/eventing/blob/0b45ad82cfd52c56e243c5766dbdf4245fd61505/pkg/reconciler/apiserversource/controller.go#L99-L102
You could do the same for the role & rolebinding